src/node_api.cc

```diff
@@ -1118,6 +1118,7 @@ napi_status NAPI_CDECL napi_get_buffer_info(napi_env env,
   CHECK_ARG(env, value);
 
   v8::Local<v8::Value> buffer = v8impl::V8LocalValueFromJsValue(value);
+  RETURN_STATUS_IF_FALSE(env, buffer->IsArrayBuffer(), napi_invalid_arg);
```

Comment:
I don't think this requires a call from C-> JavaScript so should not be too costly, what I'm not sure is if all buffers passed in must be ArrayBuffers?  @Janrupf is that documented somewhere?

Reply:
This is checked here:
https://github.com/nodejs/node/blob/09da597535583c139adf18d51af42360cbeae42f/src/node_buffer.cc#L248

Call to this method is made here:
https://github.com/nodejs/node/blob/09da597535583c139adf18d51af42360cbeae42f/src/node_api.cc#L1123

So if this check fails (and a sensible combination of output arguments is provided) this will assert when the object passed in is not a buffer.

Reply:
Actually good catch! I overlooked that the correct check would be `IsArrayBufferView`.